Mathlib/RingTheory/PowerSeries/Order.lean

```diff
@@ -317,34 +351,28 @@ theorem order_mul (φ ψ : R⟦X⟧) : order (φ * ψ) = order φ + order ψ :=
   simp only [order_eq_emultiplicity_X]
   exact emultiplicity_mul X_prime
 
--- Dividing `X` by the maximal power of `X` dividing it leaves `1`.
-@[simp]
-theorem divided_by_X_pow_order_of_X_eq_one : divided_by_X_pow_order X_ne_zero = (1 : R⟦X⟧) := by
-  rw [← mul_eq_left₀ X_ne_zero]
-  simpa using self_eq_X_pow_order_mul_divided_by_X_pow_order (@X_ne_zero R _ _)
-
--- Dividing a power series by the maximal power of `X` dividing it, respects multiplication.
-theorem divided_by_X_pow_orderMul {f g : R⟦X⟧} (hf : f ≠ 0) (hg : g ≠ 0) :
-    divided_by_X_pow_order hf * divided_by_X_pow_order hg =
-      divided_by_X_pow_order (mul_ne_zero hf hg) := by
-  set df := f.order.lift (order_finite_iff_ne_zero.mpr hf)
-  set dg := g.order.lift (order_finite_iff_ne_zero.mpr hg)
-  set dfg := (f * g).order.lift (order_finite_iff_ne_zero.mpr (mul_ne_zero hf hg))
-  have H_add_d : df + dg = dfg := by
-    simp_all [df, dg, dfg, order_mul f g]
-  have H := self_eq_X_pow_order_mul_divided_by_X_pow_order (mul_ne_zero hf hg)
-  have : f * g = X ^ dfg * (divided_by_X_pow_order hf * divided_by_X_pow_order hg) := by
-    calc
-      f * g = X ^ df * divided_by_X_pow_order hf * (X ^ dg * divided_by_X_pow_order hg) := by
-        rw [self_eq_X_pow_order_mul_divided_by_X_pow_order,
-          self_eq_X_pow_order_mul_divided_by_X_pow_order]
-      _ = X ^ df * X ^ dg * divided_by_X_pow_order hf * divided_by_X_pow_order hg := by ring
-      _ = X ^ (df + dg) * divided_by_X_pow_order hf * divided_by_X_pow_order hg := by rw [pow_add]
-      _ = X ^ dfg * divided_by_X_pow_order hf * divided_by_X_pow_order hg := by rw [H_add_d]
-      _ = X ^ dfg * (divided_by_X_pow_order hf * divided_by_X_pow_order hg) := by rw [mul_assoc]
-  refine (IsLeftCancelMulZero.mul_left_cancel_of_ne_zero (pow_ne_zero dfg X_ne_zero) ?_).symm
-  simp only [this] at H
-  convert H
+/-- The operation of dividing a power series by the largest possible power of `X`
+preserves multiplication. -/
+theorem divXPowOrder_mul {f g : R⟦X⟧} :
```

Comment:
```suggestion
theorem divXPowOrder_mul_divXPowOrder {f g : R⟦X⟧} :
```
For this direction, I think this name makes more sense; `divXPowOrder_mul` feels like it should be the name of the equality in the opposite direction.